File: C++/680 Valid Palindrome II.cpp
using namespace std;

class Solution
{
public:
    bool isPalindrome(string &s, int left, int right)
    {
        int i = left, j = right;
        while (i < j)
        {
            if (s[i] != s[j])
                return false;
            i++;
            j--;
        }
        return true;
    }

    bool validPalindrome(string s)
    {
        int n = s.size();
        int i = 0, j = n - 1;
        while (i < j)
        {
            if (s[i] != s[j])
            {
                return isPalindrome(s, i + 1, j) || isPalindrome(s, i, j - 1);
            }
            i++;
            j--;
        }
        return true;
    }
};

File: Swift/680 Valid Palindrome II.swift
class Solution {
    func isPalindrome(_ s: [String], _ l: Int, _ r: Int) -> Bool {
        var i = l, j = r
        while i < j {
            if s[i] != s[j] {
                return false
            }
            i += 1
            j -= 1
        }
        return true
    }

    func validPalindrome(_ s: String) -> Bool {
        var res: [String] = []
        for e in s {
            res.append(String(e))
        }

        var i = 0, j = res.count - 1
        while i < j {
            if res[i] != res[j] {
                return isPalindrome(res, i + 1, j) || isPalindrome(res, i, j - 1)
            }
            i += 1
            j -= 1
        }
        return true
    }
}
